src/map/battle.c

```diff
@@ -3646,6 +3646,7 @@ static struct Damage battle_calc_magic_attack(struct block_list *src, struct blo
 	flag.imdef = (nk&NK_IGNORE_DEF)? 1 : 0;
 
 	sd = BL_CAST(BL_PC, src);
+	tsd = BL_CAST(BL_PC, target);
```

Comment:
better put ``struct map_session_data *tsd = BL_CAST(BL_PC, target);`` at this line